test/SILOptimizer/di_property_wrappers.swift

```diff
@@ -535,6 +535,54 @@ func testSR_12341() {
   _ = SR_12341(condition: true)
 }
 
+class TestLeak: CustomStringConvertible {
+  let val: Int
+  init(val: Int) { self.val = val }
+  deinit { print("    .. \(self).deinit") }
+  var description: String { "TestLeak(\(val))" }
+}
+
+struct SR_13495 {
+  @Wrapper var wrapped: TestLeak
+  var str: String
+
+  init() {
+     wrapped = TestLeak(val: 42)
+     str = ""
+     wrapped = TestLeak(val: 27)
+  }
+
+  init(conditionalInit: Bool) {
+    if (conditionalInit) {
+      wrapped = TestLeak(val: 42)
+    }
+    str = ""
+    wrapped = TestLeak(val: 27)
+  }
+}
+
+func testSR_13495() {
+  // CHECK: ## SR_13495
+  print("\n## SR_13495")
+
+  // CHECK-NEXT:   .. init TestLeak(42)
+  // CHECK-NEXT:     .. TestLeak(42).deinit
+  // CHECK-NEXT:   .. set TestLeak(27)
+  // CHECK-NEXT:     .. TestLeak(27).deinit
+  _ = SR_13495()
+
+  // CHECK-NEXT:   .. init TestLeak(42)
+  // CHECK-NEXT:     .. TestLeak(42).deinit
+  // CHECK-NEXT:   .. init TestLeak(27)
```

Comment:
It seems wrong that this is doing initialization of the wrapper again, but I'm still trying to figure out how conditional "init or assign" is supposed to work

Reply:
It looks me like `DIUseKind::InitOrAssign` always turns into unconditional initialization in `LifetimeChecker::handleConditionalInitAssign()`:

```
    // Finally, now that we know the value is uninitialized on all paths, it is
    // safe to do an unconditional initialization.
    Use.Kind = DIUseKind::Initialization;
```

Reply:
Since this is an existing issue with property wrappers, I'll investigate separately